### PR TITLE
chore: provide a way to synchronise a branch of `ecobalyse-private` with a PR

### PR DESCRIPTION
**Pull request by vjousse**
## :wrench: Problem

We need a way to be able to test a PR with a specific branch of the `ecobalyse-private` repo.

## :cake: Solution

Parse the body of the PR for a line with `ecobalyse_data: <branch_name>` pattern. When deploying on scalingo, check if the current deployed commit is part of an opened PR and check if there is a `ecobalyse_data: <branch_name>` line. If so, we checkout the corresponding branch of `ecobalyse-private`.

We do the same when running the tests and score_history in the Github Workflows.

## :desert_island: How to test

Create a branch in the `ecobalyse-private`, create a PR (or use this one) with the specific line. Check that the app contains the data of the `ecobalyse-private` branch.

## :memo: Data

ecobalyse_data: organic_tomato

### CODE REVIEW

File: .github/workflows/node.js.yml [hunk 1]
```diff
@@ -76,6 +76,17 @@ jobs:
...
+    - name: Checkout specific ecobalyse-private branch
+      if: ${{ github.event_name == 'pull_request' }}
+      env:
+        LAST_COMMIT_SHA: ${{ github.event.pull_request.head.sha }}
+      run: |
+        eval `ssh-agent -s`
+        # Private ssh key used by this Github action to clone the private repo
+        # The corresponding public key must be set as a deploy key of the private repo
+        ssh-add - <<< '${{ secrets.PRIVATE_SSH_KEY }}'
+        ./bin/checkout-ecobalyse-private-branch.sh $LAST_COMMIT_SHA
```
Comment (ccomb) [inline .github/workflows/node.js.yml:88]: I understand that this is done during the build?
I created a PR with a merge of this one and the organic_tomato one (#665) to test the behaviour. I initially forgot to specify the `ecobalyse_data: <commit>` in the PR, and corrected while it was running. The test is failing by not finding a new process id which should be in the private repo. So we might need a way to detect and restart the tests/checks after modifying the description of the PR, a bit like the semantic PR title check. But would it be relevant?
```
on:
  pull_request_target:
    types:
      - opened
      - edited
      - synchronize
```

Comment (vjousse) [reply .github/workflows/node.js.yml:88]: Now it's computed on every test run. It was a bug in branch name detection, it should be fixed now.

File: buildpack-run.sh [hunk 2]
```diff
@@ -1 +1,6 @@
+#!/usr/bin/env bash
+
 git clone git@github.com:MTES-MCT/ecobalyse-private.git
+
+# Scalingo should set the $SOURCE_VERSION variable
+./bin/checkout-ecobalyse-private-branch.sh $SOURCE_VERSION
```
Comment (n1k0) [inline buildpack-run.sh:6]: What calls this bash script? I couldn't find any docs about it the Scalingo one

I really wish we could use anything else than bash and sed for advanced scripting and parsing here :/

**Review by ccomb — APPROVED** [review]
Tested at least on one case that worked :+1:

**Comment by vjousse** [comment]
Merging it now so we can synchronize branches. Refactoring using Python will come later on

**Pull request merged**